### PR TITLE
chore(tests): ensuring graphql responses don't change after launch

### PR DESCRIPTION
**Pull request by bassrock**
## Summary
Gain confidence in our test suite by shoring up it's reliability. There will be followup PRs to try and do this work iteratively.

See Implementation Details for more info.

## References 
* Links to docs, tickets, designs if available

## Implementation Details
* Ensured that we never try and change what the GraphQL can respond with after the app was launched.
    * Audited all usages of the Mock GraphQL server
    * Made it so that fallbackResponses now contain all possible baseline responses
* Ensured that where needed we were always awaiting the graphql server responses were fullfilled.
* Switched to more specific request matching
* Modified some test data to more accurately match what the server provides.
* Updated the timeouts for Snowplow assertions
    *  Also ensured that Debug builds emit analytics immediately instead of using a queue to speed up testing.
* Fixed a bad access on persistent task

## Test Steps
* Run the full test suite. See no errors

## PR Checklist:
- [ ] Added Unit / UI tests
- [ ] Self Review (review, clean up, documentation, run tests)
- [ ] Basic Self QA

### CODE REVIEW

File: PocketKit/Sources/Analytics/Tracker/PocketSnowplowTracker.swift [hunk 1]
```diff
@@ -39,6 +39,11 @@ public class PocketSnowplowTracker: SnowplowTracker {
             configurations: [trackerConfiguration]
         )
 
+        #if DEBUG || DEBUG_ALPHA_NEUE
+        // We are using a debug build, emit analytics instantly for testing instead of batches
+        tracker.emitter.bufferOption = .single
```
Comment (cyndichin) [inline PocketKit/Sources/Analytics/Tracker/PocketSnowplowTracker.swift:44]: curious on why the change?

File: Tests iOS/AddTagsItemTests.swift [hunk 2]
```diff
@@ -89,14 +72,16 @@ class AddTagsItemTests: XCTestCase {
...
         await snowplowMicro.assertBaselineSnowplowExpectation()
-        let removeTagEvent = await snowplowMicro.getFirstEvent(with: "global-nav.addTags.removeInputTag")
-        removeTagEvent!.getUIContext()!.assertHas(type: "button")
-        removeTagEvent!.getContentContext()!.assertHas(url: "http://localhost:8080/hello")
 
-        await snowplowMicro.assertBaselineSnowplowExpectation()
-        let addTagEvent = await snowplowMicro.getFirstEvent(with: "global-nav.addTags.addTag")
-        addTagEvent!.getUIContext()!.assertHas(type: "button")
-        addTagEvent!.getContentContext()!.assertHas(url: "http://localhost:8080/hello")
+        let events = await [snowplowMicro.getFirstEvent(with: "global-nav.addTags.removeInputTag"), snowplowMicro.getFirstEvent(with: "global-nav.addTags.addTag")]
```
Comment (cyndichin) [inline Tests iOS/AddTagsItemTests.swift:76]: 👍

**Review by cyndichin — APPROVED** [review]
looks good! thanks

File: Tests iOS/DeleteAnItemTests.swift [hunk 3]
```diff
@@ -80,64 +76,45 @@ class DeleteAnItemTests: XCTestCase {
...
-
                 return .delete()
-            } else {
-                fatalError("Unexpected request")
             }
+
+            return .fallbackResponses(apiRequest: apiRequest)
         }
+        app.launch()
+        app.tabBar.savesButton.wait().tap()
```
Comment (Gio2018) [inline Tests iOS/DeleteAnItemTests.swift:109]: would we ever need to also wait for `tabBar` (or any other ancestor in any other test) existence?

**Review by Gio2018 — APPROVED** [review]
Looks good @bassrock ! Test were succeeding locally and remotely, even many of the "flakey" ones. Had one question

File: Tests iOS/DeleteAnItemTests.swift (hunk 3 above)
Comment (bassrock) [reply Tests iOS/DeleteAnItemTests.swift:109]: I think waiting for the savesButton inherently waits for the tab bar but in my experience here... adding more waits does not hurt anything at all.
